compiler/src/iree/compiler/Dialect/Util/TransformOps/UtilTransformOps.cpp

```diff
@@ -8,15 +8,150 @@
 
 #include "iree/compiler/Dialect/Util/IR/UtilDialect.h"
 #include "iree/compiler/Dialect/Util/IR/UtilOps.h"
+#include "mlir/Bytecode/BytecodeWriter.h"
 #include "mlir/Dialect/Transform/IR/TransformDialect.h"
 #include "mlir/Dialect/Transform/IR/TransformOps.h"
 #include "mlir/Dialect/Transform/Interfaces/TransformInterfaces.h"
 #include "mlir/IR/Dominance.h"
 #include "mlir/IR/IRMapping.h"
+#include "mlir/Parser/Parser.h"
 #include "mlir/Transforms/DialectConversion.h"
 
 namespace mlir::iree_compiler::IREE::Util {
 
+LogicalResult
+IREE::Util::transform_dialect::CreateSerializedModuleOp::verify() {
+  if (!getBody().hasOneBlock()) {
+    return emitOpError() << "expected single block body";
+  }
+  if (getBody().front().getNumArguments() != 1) {
+    return emitOpError() << "expected body with single block argument.";
+  }
+  if (!isa<transform::TransformHandleTypeInterface>(
+          getBody().front().getArgument(0).getType())) {
+    return emitOpError()
+           << "expected body argument to be a transform op handle type";
+  }
+  if (!getRegion().empty()) {
+    for (Operation &op : getRegion().front()) {
+      if (!isa<transform::TransformOpInterface>(&op)) {
+        InFlightDiagnostic diag = emitOpError()
+                                  << "expected children ops to implement "
+                                     "TransformOpInterface";
+        diag.attachNote(op.getLoc()) << "op without interface";
+        return diag;
+      }
+    }
+  }
+  return success();
+}
+
+void IREE::Util::transform_dialect::CreateSerializedModuleOp::getEffects(
+    SmallVectorImpl<MemoryEffects::EffectInstance> &effects) {
+  transform::producesHandle(getOperation()->getOpResults(), effects);
+}
+
+DiagnosedSilenceableFailure
+IREE::Util::transform_dialect::CreateSerializedModuleOp::apply(
+    transform::TransformRewriter &rewriter,
+    transform::TransformResults &transformResults,
+    transform::TransformState &state) {
+
+  // Create a temporary module that will be erased once out of scope.
+  OwningOpRef<ModuleOp> tempModule(ModuleOp::create(getLoc()));
+
+  // Map the temporary module to the block argument of the body.
+  // This should never fail per the verifier.
+  auto scope = state.make_region_scope(getBody());
+  if (failed(state.mapBlockArguments(getBody().front().getArgument(0),
+                                     tempModule->getOperation()))) {
+    return DiagnosedSilenceableFailure::definiteFailure();
+  }
+
+  // Apply the contained ops one by one.
+  for (Operation &transform : getBody().front()) {
+    DiagnosedSilenceableFailure result =
+        state.applyTransform(cast<transform::TransformOpInterface>(transform));
+    // TODO: Support better error propagation.
+    if (result.isDefiniteFailure() || result.isSilenceableFailure()) {
+      return DiagnosedSilenceableFailure::definiteFailure();
+    }
```

Comment:
I think you can just check if this is a silenceable failure here, and continue. Otherwise, just return `result`.